src/lib/microservices/sync-service/sync-service.ts

```diff
@@ -0,0 +1,239 @@
+
+    const id = job.id;
+    let lock: any;
+
+    try {
+        lock = await database.lock(id);
+    } catch (e) {
+        logger.error(`It was not possible lock the id ${id}`, moduleName, e);
+        lock = null;
+    }
+
+    if (!lock) {
+        /*
+        * If we are not able to lock the job in the database, keep
+        * the item locked in the queue until the timeout (in the queue)
+        * expire.
+        */
+
+        return;
+    }
+
+    let error = false;
+
+    try {
+        const dbJob: IJobModel = await database.job.get(id);
+
+        if (!dbJob) {
+            logger.error(`Job ${id} not found in database`, moduleName);
+            await database.unlock(lock);
+
+            appInsightClient.trackException({ exception: new Error(`Job ${id} not found in database`) });
+
+            return;
+        }
+
+        logger.log(generateLog(`Synchronizing Job`, job, { showHint: true }), moduleName);
+
+        if (job.status === JobStatus.started) {
+            // When a job is split we receive more than one messages for the status `started`
+            // but we only want to store in the database the first one.
+            if (dbJob.status !== JobStatus.started) {
+                dbJob.webhintVersion = job.webhintVersion;
+            }
+
+            if (!dbJob.started || dbJob.started > new Date(job.started)) {
+                dbJob.started = job.started;
+            }
+
+            // double check just in case the started message is not the first one we are processing.
+            if (dbJob.status === JobStatus.pending) {
+                dbJob.status = job.status;
+            }
+        } else {
+            setHints(dbJob, job);
+
+            if (!dbJob.log) {
+                dbJob.log = '';
+            }
+
+            if (job.log) {
+                dbJob.log += `${job.log}\n`;
+            }
+
+            if (job.status === JobStatus.error) {
+                if (!dbJob.error) {
+                    dbJob.error = [];
+                }
+                dbJob.error.push(job.error);
+                await reportGithubIssues(job);
+            } else {
+                await reportGithubTimeoutIssues(job);
+            }
+
+            if (isJobFinished(dbJob)) {
+                dbJob.status = dbJob.error && dbJob.error.length > 0 ? JobStatus.error : job.status;
+
+                if (dbJob.status === JobStatus.finished) {
+                    await closeGithubIssues(dbJob);
+                }
+            }
+
+            if (!dbJob.finished || dbJob.finished < new Date(job.finished)) {
+                dbJob.finished = job.finished;
+            }
+        }
+
+        logger.log(generateLog(`Synchronized Job`, job, { showHint: true }), moduleName);
+        await database.job.update(dbJob);
+        logger.log(`Job ${id} updated in database`);
+    } catch (err) {
+        error = true;
+        logger.log(`Error updating database for Job ${id}: ${err.message}`);
```

Comment:
I think we should re-throw the exception here to get the message back to the queue and retry the sync.

Reply:
```suggestion
        logger.log(`Error updating database for Job ${id}: ${err.message}`);
        throw err;
```